src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnRpService.java

```diff
@@ -0,0 +1,269 @@
+            }
+            final String username = info.getUsername();
+            if (!StringUtils.hasText(username)) {
+                throw new WebAuthnAuthenticationException("_",
+                        "Could not finish registration: missing username");
+            }
+            WebAuthnUserAccount existingAccount = webAuthnUserAccountService.findByProviderAndUsername(provider,
+                    username);
+            if (existingAccount != null) {
+                throw new WebAuthnAuthenticationException("_",
+                        "Account already registered");
+            }
+            final PublicKeyCredentialCreationOptions options = info.getOptions();
+            RegistrationResult result = rp
+                    .finishRegistration(FinishRegistrationOptions.builder().request(options).response(pkc).build());
+            boolean attestationIsTrusted = result.isAttestationTrusted();
+            if (attestationIsTrusted) {
+                // Create user account in the database
+                WebAuthnUserAccount account = new WebAuthnUserAccount();
+                account.setUsername(username);
+                account.setRealm(realm);
+                String subject = subjectService.generateUuid(SystemKeys.RESOURCE_USER);
+                account.setSubject(subject);
+                ByteArray userHandle = info.getOptions().getUser().getId();
+                account.setUserHandle(userHandle.getBase64());
+                account.setProvider(provider);
+                webAuthnUserAccountService.addAccount(account);
+
+                // Create credential in the database
+                WebAuthnCredential newCred = new WebAuthnCredential();
+                newCred.setCreatedOn(new Date());
+                newCred.setLastUsedOn(new Date());
+                newCred.setCredentialId(result.getKeyId().getId().getBase64());
+                newCred.setPublicKeyCose(result.getPublicKeyCose().getBase64());
+                newCred.setSignatureCount(result.getSignatureCount());
+                Optional<SortedSet<AuthenticatorTransport>> transportsOpt = result.getKeyId().getTransports();
+                if (transportsOpt.isPresent()) {
+                    SortedSet<AuthenticatorTransport> transports = transportsOpt.get();
+                    final List<String> transportCodes = new LinkedList<>();
+                    for (final AuthenticatorTransport t : transports) {
+                        transportCodes.add(t.getId());
+                    }
+                    newCred.setTransports(StringUtils.collectionToCommaDelimitedString(transportCodes));
+                } else {
+                    newCred.setTransports("");
+                }
+                newCred.setUserHandle(account.getUserHandle());
+
+                webAuthnUserAccountService.saveCredential(newCred);
+                activeRegistrations.remove(key);
+                return username;
+            } else {
+                return null;
```

Comment:
remove else block, see above

Reply:
Done in previous commits (I believe it was cb95a78885936ee33c8f8d0d9ad6d34aed10c7bf)